src/main/scala/org/constellation/p2p/DataResolver.scala

```diff
@@ -170,16 +174,16 @@ class DataResolver {
   }
 
   def getReadyPeers(dao: DAO): Iterable[PeerApiClient] = {
-    dao.readyPeers.map(p => PeerApiClient(p._1, p._2.client))
+    dao.readyPeersAsync.unsafeRunSync().map(p => PeerApiClient(p._1, p._2.client))
   }
 
 }
 
 object DataResolver extends DataResolver
 
-case class DataResolutionOutOfPeers(endpoint: String,hash: String, peers: Iterable[String])
+case class DataResolutionOutOfPeers(thisNode: String, endpoint: String,hash: String, peers: Iterable[String])
     extends Exception(
-      s"Run out of peers when resolving: $endpoint with hash: $hash following tried: $peers"
+      s"node [${thisNode}] Run out of peers when resolving: $endpoint with hash: $hash following tried: $peers"
```

Comment:
What's the value of the `endpoint` val typtically? I just found `val endpoint: String = "endpoint"`. Extrapolating on the style, I'd expect `resolving: endpoint $endpoint` here. 

Sorry for the superficial notes, I'll get deeper when I've an endpoint to quiery.

Reply:
This change is only about adding debugging info. We're kinda unable to think about and fix all places we're touching :)

Reply:
I know the feel

Reply:
endpoint will be either `checkpoint` or `transaction`